### ISSUE TITLE
False-positive void-return in dtslint after adding awaited

### ISSUE THREAD

**Issue by Maxim-Mazurok** [microsoft/TypeScript#37551]
**TypeScript Version:**  from 3.9.0-dev.20200321 to 3.9.0-dev.20200324 

**Code**

```ts
declare namespace gapi.client {
    function load(name: "abusiveexperiencereport", version: "v1"): PromiseLike<void>;
    function load(name: "abusiveexperiencereport", version: "v1", callback: () => any): void;
}
```

**Expected behavior:**
[`dtslint`](https://github.com/microsoft/dtslint) uses rule [`void-return`](https://github.com/microsoft/dtslint/blob/master/docs/void-return.md) which didn't fail for the above code and shouldn't fail.

**Actual behavior:**
When `dtslint` is paired with `typescript` version since this commit: https://github.com/microsoft/TypeScript/commit/e3ec7b18b82d28f796ec635267907ff9014c4c5b which was merged in https://github.com/microsoft/TypeScript/pull/35998 - it fails:
```
ERROR: 17:80  void-return  Use the `void` type for return types only. Otherwise, use `undefined`. See: https://github.com/Microsoft/dtslint/blob/master/docs/void-return.md
ERROR: 18:89  void-return  Use the `void` type for return types only. Otherwise, use `undefined`. See: https://github.com/Microsoft/dtslint/blob/master/docs/void-return.md
```

----

Related to https://github.com/Maxim-Mazurok/google-api-typings-generator/pull/101

**Comment by Maxim-Mazurok** [comment]
Sorry, it seems like a `dstlint` issue actually.
It's using bundled TS (`node_modules/dtslint/nodemodules/typescript`) for rules, but project's TS (`node_modules/typescript`) for parsing sources before running rules on them. 
So, node kinds reported by older TS version are not matching with the new ones. That's what was causing problems, as node kinds were updated in this PR.
Probably it would be great to have backward-compatibility in this matter, so I'm not closing the issue.

**Comment by RyanCavanaugh** [comment]
We don't support this kind of back compat

**Comment by Maxim-Mazurok** [comment]
@RyanCavanaugh I hear you. But what if we could support it?
From what I understand, it would require changing https://github.com/microsoft/TypeScript/blob/a1c8608f681488fda3f0b6ffd89195a81f80f0b0/src/compiler/types.ts#L124-L127 to 
```ts
export const enum SyntaxKind { 
     Unknown = "Unknown", 
     EndOfFileToken = "EndOfFileToken", 
     SingleLineCommentTrivia = "SingleLineCommentTrivia",
```
Kind of like it was did in https://github.com/microsoft/TypeScript/blob/a1c8608f681488fda3f0b6ffd89195a81f80f0b0/src/compiler/types.ts#L4207-L4210

Another approach is to do this like for different flags https://github.com/microsoft/TypeScript/blob/a1c8608f681488fda3f0b6ffd89195a81f80f0b0/src/compiler/types.ts#L4163-L4166 so it will look like 
```ts
export const enum SyntaxKind { 
     Unknown                 = 1 << 0, 
     EndOfFileToken          = 2 << 0, 
     SingleLineCommentTrivia = 3 << 0,
```
or even simpler:
```ts
export const enum SyntaxKind { 
     Unknown                 = 0, 
     EndOfFileToken          = 1, 
     SingleLineCommentTrivia = 2,
```

Any of these approaches will preserve enum values and make TS even more back compat :) What do you think?

**Comment by Maxim-Mazurok** [comment]
@rbuckton what do you think about https://github.com/microsoft/TypeScript/issues/37551#issuecomment-603635133 ?
I think if we use this approach - we can eliminate potential issues when updating `SyntaxKind` in the future.
